src/usr/local/sbin/openvpn.learn-address.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+# openvpn learn-address script maintaining DNS entries of connected clients in
+# unbound config.
+
+DOMAIN="$1"
+OP="$2"
+IP="$3"
+CN="$4"
+
+DIR="/var/unbound"
+CONF="$DIR/openvpn.client.$IP.conf"
+PIDFILE="/var/run/unbound.pid"
+
+case "$OP" in
+
+	add|update)
+		# Remove all configs which mention the FQDN
+		grep -l -null -F "local-data: \"$CN.$DOMAIN A $IP\"" $DIR/openvpn.client.*.conf | xargs -0 rm
+		rm -f "$CONF"
+
+		TMPCONF=$(mktemp "$CONF.XXXXXX")
+		TMPSRV=$(mktemp "$CONF.XXXXXX")
+
+		# Add new local-data entry.
+		(
+			echo "local-data-ptr: \"$IP $CN.$DOMAIN\"" &&
+			echo "local-data: \"$CN.$DOMAIN A $IP\"" &&
+			echo "local-data: \"$CN A $IP\""
+		) > "$TMPCONF"
+
+		# Check syntax, install configuration and restart unbound.
+		(
+			echo "server:" &&
+			echo "chroot: $DIR" &&
+			echo "directory: $DIR" &&
+			echo "include: $TMPCONF"
+		) > "$TMPSRV"
+
+		chmod 644 "$TMPCONF" "$TMPSRV"
+		/usr/local/sbin/unbound-checkconf "$TMPSRV" && mv "$TMPCONF" "$CONF"
+		rm -f "$TMPCONF" "$TMPSRV"
```

Comment:
At the very least, a test to make sure that both parameters have some value that (e.g. not empty) and that none of them are only '/'. Ideally, it should also test if the file exists before trying to use `rm -rf` on it rather than assuming it does. There are a couple other usages of rm in the script that could use the same treatment.

Basically, we need to make sure the script won't `rm -rf` the entire filesystem or more files than intended, especially if somehow the mktemp call does not return an expected result.

tl;dr: Make no assumptions, assume input is invalid until proven otherwise, especially when removing files using variable names.